.semaphore/semaphore.yml

```diff
@@ -0,0 +1,139 @@
+    processing: parallel
+global_job_config:
+  env_vars:
+    - name: NO_OF_TEST_WORKER_NODES
+      value: "2"
+    - name: MAX_TEST_RETRY_FAILURES_PER_NODE
+      value: "8"
+    - name: TEST_TIMEOUT_IN_MINUTES
+      value: "240"
+    - name: NANO_VERSION
+      value: "true"
+    - name: PUBLISH
+      value: "true"
+    - name: ENABLE_UPDATE_VERSION
+      value: "true"
+    - name: ENABLE_PUBLISH_ARTIFACTS
+      value: "true"
+    - name: ENABLE_DOWNSTREAM_TRIGGER
+      value: "true"
+    - name: ENABLE_PROPAGATE_CC_RELEASE_BRANCH
+      value: "true"
+    - name: DOWNSTREAM_PROJECTS
+      value: "common"
+  prologue:
+    commands:
+      - echo $SEMAPHORE_WORKFLOW_ID
+      - echo $SEMAPHORE_GIT_REPO_SLUG
+      - checkout
+      - . vault-setup
+      - . vault-sem-get-secret ci-reporting
+      - . vault-sem-get-secret sox-semaphore-build-info
+      - sem-version java 21
+      - sem-version go 1.16.15
+      - git config --global url."git@github.com:".insteadOf "https://github.com/"
+      - export SEMAPHORE_CACHE_DIR=/home/semaphore
+      - source scripts/set_env_vars.sh
+      - source scripts/set_downstream_branch.sh
+blocks:
+  - name: Tests
+    dependencies: []
+    task:
+      prologue:
+        commands:
+          - make init-ci
+      jobs:
+        - name: Unit tests and Integration tests
+          commands:
+            - './gradlew 
+                unitTest integrationTest --no-daemon --stacktrace --continue 
+                -PtestLoggingEvents=started,passed,skipped,failed -PmaxParallelForks=4 
+                -PignoreFailures=true -PmaxTestRetries=1 -PmaxTestRetryFailures=5'
+          execution_time_limit:
+            minutes: 120
+      epilogue:
+        always:
+          commands:
+            - shopt -s globstar && test-results publish --name Test-Suite --trim-output-to 1024 --omit-output-for-passed **/build/test-results/**/TEST-*.xml
+            - |
+              echo "Job creation time: $((SEMAPHORE_JOB_CREATION_TIME * 1000))"
+              echo "Current time: $(date +%s%3N)"
+              echo $(( ($(date +%s%3N) - $SEMAPHORE_JOB_CREATION_TIME * 1000) )) > duration.txt
+              artifact push workflow duration.txt --destination $SEMAPHORE_JOB_INDEX-test-duration-$SEMAPHORE_WORKFLOW_ID || true
+
+  - name: Gradle Build
+    dependencies: []
+    task:
+      prologue:
+        commands:
+          - make init-ci
+      jobs:
+        - name: Build, Compile, Validations, Publish, Upgrade
+          commands:
+            - >-
+              if [ "$NANO_VERSION" = "true" ] && [ "$RELEASE_JOB" = "false" ] && [ "$ENABLE_PUBLISH_ARTIFACTS" = "true" ]; then
+                . ci-tools ci-update-version
+              fi
+            - make compile-validate
+            - make check-scala-compatibility
+            - |
+              if [ "$PUBLISH" = "true" ] && [ "$SEMAPHORE_GIT_REF_TYPE" != "pull-request" ] && [ "$ENABLE_PUBLISH_ARTIFACTS" = "true" ]; then \
```

Comment:
ensure these conditions are proper and relevant.

Reply:
yes, it maps directly to 
```
if (config.publish) {
      stage("Publish to artifactory") {
        if (!config.isReleaseJob && !config.isPrJob) {
            ciTool("ci-push-tag ${env.WORKSPACE} kafka")
        }

        if (config.isDevJob) {
          publishStep('artifactory_snapshots_settings')
        } else if (config.isPreviewJob) {
          publishStep('artifactory_preview_release_settings')
        }
      }
    }
```
from jenkinsfile